src/r2pipe.rs

```diff
@@ -127,32 +133,24 @@ impl R2Pipe {
     pub fn open() -> Result<R2Pipe> {
         unimplemented!()
     }
-
-    pub fn cmd(&mut self, cmd: &str) -> Result<String> {
+    fn get_piper<'a>(&'a mut self) -> &'a mut dyn Piper {
         match *self {
-            R2Pipe::Pipe(ref mut x) => x.cmd(cmd.trim()),
-            R2Pipe::Lang(ref mut x) => x.cmd(cmd.trim()),
-            R2Pipe::Tcp(ref mut x) => x.cmd(cmd.trim()),
-            R2Pipe::Http(ref mut x) => x.cmd(cmd.trim()),
+            R2Pipe::Pipe(ref mut x) => x,
+            R2Pipe::Lang(ref mut x) => x,
+            R2Pipe::Tcp(ref mut x) => x,
+            R2Pipe::Http(ref mut x) => x,
         }
     }
+    pub fn cmd(&mut self, cmd: &str) -> Result<String> {
+        self.get_piper().cmd(cmd.trim())
+    }
 
     pub fn cmdj(&mut self, cmd: &str) -> Result<Value> {
-        match *self {
-            R2Pipe::Pipe(ref mut x) => x.cmdj(cmd.trim()),
-            R2Pipe::Lang(ref mut x) => x.cmdj(cmd.trim()),
-            R2Pipe::Tcp(ref mut x) => x.cmdj(cmd.trim()),
-            R2Pipe::Http(ref mut x) => x.cmdj(cmd.trim()),
-        }
+        self.get_piper().cmdj(cmd.trim())
```

Comment:
Can't this be written as `self.cmdj(cmd.trim())` and we can remove the `get_piper` function?

Reply:
I don't think so, self.cmdj calls the cmdj method defined on the R2Pipe struct and not the Pipe trait but I am not good with rust so I can't tell.